xbmc/threads/platform/RecursiveMutex.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+ *  Copyright (C) 2005-2022 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#include "RecursiveMutex.h"
+
+namespace XbmcThreads
+{
+
+static pthread_mutexattr_t recursiveAttr;
+
+static bool SetRecursiveAttr()
+{
+  static bool alreadyCalled = false;
+
+  if (!alreadyCalled)
+  {
+    pthread_mutexattr_init(&recursiveAttr);
+    pthread_mutexattr_settype(&recursiveAttr, PTHREAD_MUTEX_RECURSIVE);
+#if !defined(TARGET_ANDROID)
+    pthread_mutexattr_setprotocol(&recursiveAttr, PTHREAD_PRIO_INHERIT);
+#endif
+    alreadyCalled = true;
+  }
+
+  return true; // note, we never call destroy.
+}
+
+static bool recursiveAttrSet = SetRecursiveAttr();
+
+pthread_mutexattr_t* CRecursiveMutex::getRecursiveAttr()
```

Comment:
I know this signature comes from existing code, but is there any merit to returning `pthread_mutexattr_t&` to indicate that `nullptr` can't be returned?

Reply:
I added this change as another commit so it's obvious what was changed.